### ISSUE TITLE
Strong slowdown in the recent versions

### ISSUE THREAD

**Issue by sedyh** [yohamta/donburi#109]
### Problem

Hello, I found a very strange behavior. Looks like the version of lib that is a year older works better than the current one...

Branch `develop`:
![image](https://github.com/yohamta/donburi/assets/19890545/fb7e7e34-0ecf-4d68-b3d4-25d699c0f2cd)

Branch `main`:
![image](https://github.com/yohamta/donburi/assets/19890545/c2138a42-96ad-4a13-b51e-1d14a91faf7a)

Also, latest tag has broken wheel in bunnymark example.

**Comment by yohamta** [comment]
Hi 👋 Yeah, that seems weird. I'll give it a test to see if I can replicate the issue.

**Comment by yohamta** [comment]
I've also run bunnymark on my M2 mac and it's true that the performance has declined. It seems one of my recent refactoring may be the cause which I need to identify the commit.

| Commit  | Max entities per 60 fps |
| ------------- | ------------- |
| 4b9bfbd  | 52000 [screenshot](https://github.com/yohamta/donburi/assets/1475839/92560b38-6fd0-46b0-b451-c4570a4d4c51)  | 
| 4b9bfbd  | 65000 [screenshot](https://github.com/yohamta/donburi/assets/1475839/28a80016-1f14-4d09-a0df-f51ecead0e99)  |

**Comment by sedyh** [comment]
Hello, is there any updates on this?

**Comment by yohamta** [comment]
Hello, not right now. I'm thinking that there might be the other way to improve the performance. Maybe we can optimize the component data storage for memory alignment. Not sure how much it works until experiment.

**Comment by setanarut** [comment]
Please specify the previous and next donburi version numbers. Which version is faster. Which version is slower?

**Comment by sedyh** [comment]
> Please specify the previous and next donburi version numbers.

Hello, they were listed above.

https://github.com/yohamta/donburi/issues/109#issuecomment-1752280935

**Comment by setanarut** [comment]
> > Please specify the previous and next donburi version numbers.
> 
> Hello, they were listed above.

> 
> [#109 (comment)](https://github.com/yohamta/donburi/issues/109#issuecomment-1752280935)

These are not version numbers. The version number does not appear in the Github Mobile app, that's why I sent a message.

1. [v1.3.7](https://github.com/yohamta/donburi/releases/tag/v1.3.7) - Fast
2. [v1.3.8](https://github.com/yohamta/donburi/releases/tag/v1.3.8) - Slow

**Comment by sedyh** [comment]
> These are not version numbers. The version number does not appear in the Github Mobile app, that's why I sent a message.

Yes, this is approximately 1.3.7 and 1.3.8, but I do not advise to tie into versions, cause specific commits were tested. From the point of view of go get, both can be specified via @.
